packages/ses/src/module-load.js

```diff
@@ -175,27 +175,64 @@ const loadWithoutErrorAnnotation = async (
     )} in compartment ${q(compartment.name)}`;
   }
 
-  if (staticModuleRecord.record !== undefined) {
-    const {
-      compartment: aliasCompartment = compartment,
-      specifier: aliasSpecifier = moduleSpecifier,
-      record: aliasModuleRecord,
-      importMeta,
-    } = staticModuleRecord;
+  // check if record is a RedirectStaticModuleInterface
+  if (staticModuleRecord.specifier !== undefined) {
+    // check if this redirect with an explicit record
+    if (staticModuleRecord.record !== undefined) {
+      // ensure expected record shape
+      if (staticModuleRecord.compartment !== undefined) {
+        throw new TypeError(
+          'Cannot redirect to an explicit record with a specified compartment',
+        );
+      }
+      const {
+        compartment: aliasCompartment = compartment,
+        specifier: aliasSpecifier = moduleSpecifier,
+        record: aliasModuleRecord,
+        importMeta,
+      } = staticModuleRecord;
 
-    const aliasRecord = await loadRecord(
-      compartmentPrivateFields,
-      moduleAliases,
-      aliasCompartment,
-      aliasSpecifier,
-      aliasModuleRecord,
-      pendingJobs,
-      moduleLoads,
-      errors,
-      importMeta,
+      const aliasRecord = await loadRecord(
+        compartmentPrivateFields,
+        moduleAliases,
+        aliasCompartment,
+        aliasSpecifier,
+        aliasModuleRecord,
+        pendingJobs,
+        moduleLoads,
+        errors,
+        importMeta,
+      );
+      mapSet(moduleRecords, moduleSpecifier, aliasRecord);
+      return aliasRecord;
```

Comment:
This is fine. But, reviewing this has given me a set of fresh eyes and there’s an existing defect. Having the memo set after await makes suspicious that there might be a race condition here. 

But, `loadRecord`, on closer inspection, is synchronous! We should get rid of `await` in `await loadRecord` and remove `async` from the `loadRecord` declaration.

There remains a possibility here that concurrent calls to `importHook` that redirect to the same `specifier` with a `record` will race to define that module (the last one winning), and then an `importHook` for the aliased specifier might load it again. Fixing _that_ problem is definitely out of scope for this change. The proper solution to something like that is probably to thread a cancellation token thru `importHook` and ensure the _first_ attempt to define the aliased module wins, aborting and ignoring any that remain. (Way out of scope. I’ll make an issue.)